backend/dataall/base/db/exceptions.py

```diff
@@ -181,3 +181,26 @@ def __init__(self, action, message):
 
     def __str__(self):
         return f'{self.message}'
+
+
+class ResourceThresholdExceeded(Exception):
+    def __init__(self, username, action):
+        self.username = username
+        self.action = action
+        self.message = f"""
+                    An error occurred (ResourceThresholdExceeded) when calling {self.action} operation:
+                    Requests exceeded max daily invocation count for User: {self.username}
+                """
+
+    def __str__(self):
+        return f'{self.message}'
+
+
```

Comment:
ModelGuardrailException is used only once in `dataall.modules.worksheets.services.worksheet_service.WorksheetService.analyze_text_genai`, cannot we just raise a regular Exception?

Reply:
we can ... but thinking ahead if data.all was to support additional genAI features that were invoking models (ex: automated metadata generation) this custom exception could be re-used and be more helpful imo

Reply:
Convinced :)